ckanext/iati_generator/actions/iati.py

```diff
@@ -0,0 +1,92 @@
+import logging
+import csv
+
+from ckan.plugins import toolkit
+
+from ckanext.iati_generator.csv import row_to_iati_activity
+from ckanext.iati_generator.utils import generate_final_iati_xml, get_resource_file_path
+
+
+log = logging.getLogger(__name__)
+
+# Limit the number of rows to process to avoid large XML files
+ROWS_LIMIT = int(toolkit.config.get("ckanext.iati_generator.rows_limit", 50000))
+MAX_ALLOWED_FAILURES = int(toolkit.config.get("ckanext.iati_generator.max_allowed_failures", 10))
```

Comment:
Do not execute non-trivial code in the root of a Python module. This'll run every time we read the file
Move this to the function or create a external function which is only called within the function.